components/main-chef-wrapper/lib/version.go

```diff
@@ -0,0 +1,118 @@
+package lib
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"io/ioutil"
+	"encoding/json"
+	"github.com/mnogu/go-dig"
```

Comment:
This looks helpful, but I'm concerned because it looks like it was created over the course of 1-2 days and has not received any attention since then.

Reply:
I think I saw some talk around defining a nested struct to unmarshal into.  While we can't do that for the gem version manifest, is it possible to do it for the workstation versoin manifest? Or is the structure of that file problematic, eg there's no array of definitions under `software`, but rather a bunch of named json objects there?